lib/sqlalchemy/dialects/sqlite/base.py

```diff
@@ -2624,15 +2624,16 @@ def get_check_constraints(self, connection, table_name, schema=None, **kw):
             connection, table_name, schema=schema, **kw
         )
 
-        CHECK_PATTERN = r"(?:CONSTRAINT (.+) +)?" r"CHECK *\( *(.+) *\),? *"
+        # Notes:
+        # * The pattern currently matches any character for the name of the constraint
+        #   except tab and newline characters.
+        # * Because check constraints in the table data can contain space, newline and tab characters,
+        #   the pattern matches any character untill either the beginning of the next CONSTRAINT
+        #   statement using a non-capturing non-consuming group (allowing the next one to match),
+        #   or the end of the table definition e.g. newline and closing ')'.
+        CHECK_PATTERN = r"(?:CONSTRAINT ([^\t\n]+) )?CHECK \((.+?)\)(?:, *\n\t?(?=CONSTRAINT|CHECK)|\n\))"
```

Comment:
I don't understand why there is a need to change the first part of the regexp. tab is valid inside a constraint name:
```sql
create table foo (a int, constraint "foo\tbar" CHECK (a>0))
```

Reply:
newline are too
```sql
create table foo (a int, constraint "foo\nbar" CHECK (a>0))
```

Reply:
Yes, this replaces https://github.com/sqlalchemy/sqlalchemy/pull/11678, which can be closed.

Regarding the constraint names; you are right, tabs and newlines are allowed in constraint names.
The reason for this change was to prevent multiple constraint definitions from being eagerly matched by the initial fix in https://github.com/sqlalchemy/sqlalchemy/pull/11678.
But with our added non-capturing non-consuming group to mark the end of the constraint definition, the original name part can be reinstated. We'll readjust this.

Reply:
We reinstated the name part in https://github.com/sqlalchemy/sqlalchemy/pull/11701/commits/e98b7522240c5c8f7f8698328eaf76e7566448d5, but made the group non-greedy to prevent the previously mentioned issue.